app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -117,6 +136,7 @@ class AddAdditionalMetadataController @Inject() (
       .saveMetadata(consignmentId, fileIds, request.token.bearerAccessToken, updateMetadataInputs)
       .map(_ => Redirect(routes.AdditionalMetadataSummaryController.getSelectedSummaryPage(consignmentId, metadataType, fileIds)))
   }
+  // scalastyle:on cyclomatic.complexity
```

Comment:
I know you didn't write this but it's going to need fixing. The call to `customMetadataService.deleteMetadata` on line 133 returns a Future but it isn't mapped to anything. If the http call completes before this has finished for whatever reason, we're going to get some weird behaviour.

I tried a couple of different bits to get it mapped with the if statement, this is one of them
```scala
    for {
      _ <- customMetadataService
        .saveMetadata(consignmentId, fileIds, request.token.bearerAccessToken, updateMetadataInputs)
      _ <-
        if (deleteMetadataNames.nonEmpty) {
          customMetadataService.deleteMetadata(fileIds, request.token.bearerAccessToken, deleteMetadataNames)
        } else {
          Future.successful()
        }
    } yield Redirect(routes.AdditionalMetadataSummaryController.getSelectedSummaryPage(consignmentId, metadataType, fileIds))
```
Something like that anyway.

Reply:
Good spot. Shouldn't it be the other way around though, `customMetadataService.deleteMetadata` before the `customMetadataService.saveMetadata` to ensure that we remove properties and dependencies before?

Reply:
Good point yeah, the delete should be first :+1: